src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1272,98 +1323,95 @@ void CouplingScheme::computeTimeStep(real &dt)
-        !mesh2.m_elemData.m_is_element_thickness_set )
-   {
-      return; 
-   }
-
-   parameters_t & parameters = parameters_t::getInstance();
-   real proj_ratio = parameters.timestep_pen_frac;
-   ContactPlaneManager& cpMgr = ContactPlaneManager::getInstance();
-   int dim = this->spatialDimension();
-   int numNodesPerCell1 = mesh1.m_numNodesPerCell;
-   int numNodesPerCell2 = mesh2.m_numNodesPerCell;
-
-   // Loop over interface pairs and check existing gaps for pairs in 
-   // contact, and perform velocity projection check for all contact 
-   // candidates.
-   IndexType numPairs = m_interfacePairs->getNumPairs();
-   real dt_temp1 = dt;
-   real dt_temp2 = dt;
-   int cpID = 0; 
-   bool exceed_max_gap1 = false;
-   bool exceed_max_gap2 = false;
-   bool neg_dt_gap_msg = false;
-   bool neg_dt_vel_proj_msg = false;
-   for (IndexType kp = 0; kp < numPairs; ++kp)
-   {
-      InterfacePair pair = m_interfacePairs->getInterfacePair(kp);
-
-      // guard against the case where a pair does not have a contact plane; that is, 
-      // the pair is not a contact candidate
-      if (!pair.isContactCandidate)
-      {
-         continue;
-      }
-
-      real x1[dim * numNodesPerCell1];
-      real v1[dim * numNodesPerCell1];
-      mesh1.getFaceCoords( pair.pairIndex1, &x1[0] );
-      mesh1.getFaceNodalVelocities( pair.pairIndex1, &v1[0] );
-
-      real x2[dim * numNodesPerCell2];
-      real v2[dim * numNodesPerCell2];
-      mesh2.getFaceCoords( pair.pairIndex2, &x2[0] );
-      mesh2.getFaceNodalVelocities( pair.pairIndex2, &v2[0] );
+  // note: the timestep vote is based on a maximum allowable interpenetration
+  // approach checking current gaps and then performing a velocity projection.
+  // This timestep vote is not derived from a stability analysis and does not 
+  // account for the spring stiffness in a CFL-like timestep constraint. 
+  // The timestep vote in this routine is intended to avoid missing contact or 
+  // inadequately resolving contact by detecting 'too much' interpenetration. 
+  // To do this, the first check is a 'gap check' where the current gap is 
+  // checked against a fraction of the element-thicknesses. The second check 
+  // assumes zero gap, and then does a velocity projection followed by the 
+  // 'gap check' using this new, projected gap.
+
+  auto& mesh1 = getMesh1();
+  auto& mesh2 = getMesh2();
+ 
+  // check if the element thicknesses have been set. This is now
+  // regardless of exact penalty calculation type, since element
+  // thicknesses are required for auto contact even if a constant
+  // penalty is used
+  if ( !mesh1.getElementData().m_is_element_thickness_set ||
+       !mesh2.getElementData().m_is_element_thickness_set )
+  {
+    return; 
+  }
+
+  RealT proj_ratio = m_parameters.timestep_pen_frac;
+  //int num_sides = 2; // always 2 sides in a single coupling scheme
+  int dim = spatialDimension();
+
+  // loop over each interface pair. Even if pair is not in contact, 
```

Comment:
Update the comment here to indicate that we are looping over contact planes. To avoid confusion, include the fact that if a contact plane exists, then it is a contact candidate and for this calculation we want to compute a timestep for all candidates, not necessarily ones that are deemed to be in contact per the gap constraint.